src/main/java/com/exasol/adapter/request/parser/PushdownSqlParser.java

```diff
@@ -192,8 +198,30 @@ private SqlNode parseLiteralBool(final JsonObject expression) {
         return sqlNodes;
     }
 
-    private SqlGroupBy parseGroupBy(final JsonArray groupBy) {
-        return groupBy == null ? null : new SqlGroupBy(parseExpressionList(groupBy));
+    private SqlGroupBy parseGroupBy(final JsonObject select, final SqlSelectList selectList) {
+        if (hasSingleGroupAggregation(select) && !hasAggregateFunction(selectList)) {
+            // If the aggregationType is single_group and there is no an aggregate function,
+            // we limit the result to a single row.
+            return new SqlGroupBy(List.of(new SqlLiteralString("a")));
```

Comment:
I think that functionallity does not belong into the parser. As far as I unsersdtand, it's alreaddy an implementation for handling these types of groups. This structure should in ,y oppinion be a representation of the JSON format. Different virtual schemas can then handle it differently.
You could so this transformation in a sepearate class called for example `SingleGroupTransformter`